Plugins/ActSVG/src/PortalSvgConverter.cpp

```diff
@@ -0,0 +1,159 @@
+        Acts::ActsScalar r = Acts::VectorHelpers::perp(refPosition);
+        position = Acts::Vector3(r * std::cos(refC), r * std::sin(refC),
+                                 refPosition.z());
+      } else {
+        throw std::invalid_argument("convertMultiLink: incorrect binning.");
+      }
+      Acts::Vector3 direction = surface.normal(gctx, position);
+      pLinks.push_back(makeProtoLink(portalOptions, position,
+                                     Acts::Vector3(sign * direction), v));
+    }
+  }
+  return pLinks;
+}
+
+}  // namespace
+
+Acts::Svg::ProtoPortal Acts::Svg::PortalConverter::convert(
+    const GeometryContext& gctx, const Experimental::Portal& portal,
+    const PortalConverter::Options& portalOptions) {
+  ProtoPortal pPortal;
+  // First convert the surface
+  pPortal._surface = SurfaceConverter::convert(gctx, portal.surface(),
+                                               portalOptions.surfaceOptions);
+
+  // Reference point and direction
+  Vector3 rPos(0., 0., 0);
+  Vector3 rDir(0., 0., 1);
+  const auto& surface = portal.surface();
+  const auto surfaceTransform = portal.surface().transform(gctx);
+  const auto surfaceTranslation = surfaceTransform.translation().eval();
+  const auto surfaceType = surface.bounds().type();
+  const auto& boundValues = surface.bounds().values();
+  switch (surfaceType) {
+    case SurfaceBounds::eCylinder: {
+      // Get phi
+      ActsScalar r = boundValues[0u];
+      ActsScalar aphi = boundValues[3u];
+      rPos = Vector3(r * std::cos(aphi), r * std::sin(aphi),
+                     surfaceTranslation.z());
+    } break;
+    case SurfaceBounds::eDisc: {
+      // Get phi
+      ActsScalar r = 0.5 * (boundValues[0u] + boundValues[1u]);
+      ActsScalar aphi = boundValues[3u];
+      rPos = Vector3(r * std::cos(aphi), r * std::sin(aphi),
+                     surfaceTranslation.z());
+    } break;
+    default: {
+      rPos = surfaceTranslation;
+    } break;
+  }
+  rDir = surface.normal(gctx, rPos);
+
+  // Now convert the link objects
+  const auto& updators = portal.detectorVolumeUpdators();
+
+  int sign = -1;
+  for (const auto& dvu : updators) {
+    // Get the instance and start the casting
+    const auto* instance = dvu.instance();
+    auto singleLink =
+        dynamic_cast<const Experimental::SingleDetectorVolumeImpl*>(instance);
+    if (singleLink != nullptr) {
+      pPortal._volume_links.push_back(makeProtoLink(
+          portalOptions, rPos, Vector3(sign * rDir), singleLink->dVolume));
+    }
+    auto multiLink =
+        dynamic_cast<const Experimental::BoundVolumesGrid1Impl*>(instance);
+    if (multiLink != nullptr) {
+      auto pLinks = convertMultiLink(gctx, *multiLink, surface, rPos,
+                                     portalOptions, sign);
+
+      pPortal._volume_links.insert(pPortal._volume_links.end(), pLinks.begin(),
+                                   pLinks.end());
+    }
+    // Switch to the other side
+    sign += 2;
```

Comment:
should we somehow guarantee that this can run only a single time?

Reply:
At the moment this is safe, there is an array of size 2 over which it runs, I would like to address the storage and addressing of the `Portal::Forward`, `Portal::Backward` attachment though and not leave it in an implicit `std::array<X,2u>` but that would be in another PR unrelated to SVG.